crates/block-producer/src/errors.rs

```diff
@@ -113,7 +113,7 @@ impl From<AddTransactionError> for tonic::Status {
             )
             | AddTransactionError::Expired { .. }
             | AddTransactionError::TransactionDeserializationFailed(_) => {
-                Self::invalid_argument(value.to_string())
+                Self::invalid_argument(value.as_report())
```

Comment:
This is the magic sauce to return an more indepth error via gRPC. `to_string` just provides the top level error, whereas now we return a full error report.

I haven't really tested this in-depth. So need to keep an eye out. The faucet should still error out now even if this is wrong, so it won't go undetected.